### PR TITLE
Create the edit and approve page

### LINKED ISSUE #98: create the edit and approve page
this page should be reached by clicking the "Edit and Approve" button on the prospects page. it should retrieve the prospect from the API, fill the form's details, and update the prospect.

[figma comps](https://www.figma.com/file/J3D7KOV97p0EsykJUCDL7U/Curation-Tools?node-id=1498%3A30&viewport=0%2C0%2C1) for layout reference.

### PR DESCRIPTION
**Pull request by nina-py**
- Added form validation to Edit & Approve form

- Added required fields as discussed: headline, excerpt, topic

- Added current list of topics to select dropdown

- Added tests both for the form component and the page

Closes #98.

## Implementation Decisions

Note that this is still rough around the edges but since it was getting pretty big I thought I shouldn't be adding any more to it for the time being.

I have wired this up both to the Add Story form - so you can immediately edit the prospect data parsed by the API, as well as the "Edit & Approve" button on the Prospects tab. If you access the page directly, the prospect won't be loaded from the API - I think this should be done later on, once the API is reworked otherwise multiple calls to Apollo directly from the page get very messy very quickly.

For the same reason (waiting for API abstraction layer), only the "Approve" button sends a call to the API at the moment - the other buttons just show a success message regardless. Additionally, a successful update sends the user back to the tab they came from The loading spinner remains for now - to be updated to a consistent UI further down the track.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -88,7 +88,7 @@ function App(): JSX.Element {
                   </Route>
                   <Route
                     exact
-                    path="/:feed/prospects/article/edit-and-approve/:articleid/"
+                    path="/:feed/prospects/article/:articleid/edit-and-approve/"
```
Comment (kkyeboah) [inline src/App.tsx]: Could we update the URL path structure? What do you think about this suggestion?
```suggestion
                    path="/:feed/prospects/:prospectId/edit-and-approve/"
```

File: src/pages/EditAndApproveStoryPage/EditAndApproveStoryPage.tsx [hunk 2]
```diff
@@ -9,15 +30,131 @@ interface EditAndApproveStoryPageProps {
...
+    ApproveProspectData,
+    ApproveProspectVariables
+  >(approveProspect);
+
+  /**
+   * Collect form data, choose action and send it off to the API
+   *
+   * @param data
+   */
+  const handleSubmit = (data: EditAndApproveStoryFormData) => {
```
Comment (kkyeboah) [inline src/pages/EditAndApproveStoryPage/EditAndApproveStoryPage.tsx:56]: Could we abstract the graphql query from this component as we discussed? 
Ideally, in this component, the `handle` method would be calling functions like `approveProspect` and `snoozeProspect` with arguments.

Let me know if the plan is to do that in a follow-up PR, that works for me as well.

**Review by kkyeboah — COMMENTED** [review]
Good work overall, I just had some suggestions and questions

File: src/App.tsx (hunk 1 above)
Comment (nina-py) [reply src/App.tsx]: Great suggestion, let me update the code...

File: src/pages/EditAndApproveStoryPage/EditAndApproveStoryPage.tsx (hunk 2 above)
Comment (nina-py) [reply src/pages/EditAndApproveStoryPage/EditAndApproveStoryPage.tsx:56]: Yes, the current plan is to refactor API calls in frontend code in smaller follow-up PRs: https://github.com/Pocket/curation-tools-frontend/issues/99#issuecomment-766624513. I would definitely prefer this PR to be merged more or less as is and address the imperfections in subsequent PRs.

File: src/App.tsx (hunk 1 above)
Comment (nina-py) [reply src/App.tsx]: Done! For consistency, would you like me to update the URL for the "Add Story" page as well? At the moment it's `/:feed/prospects/article/add/`.

File: src/pages/AddStoryPage/AddStoryPage.tsx [hunk 3]
```diff
@@ -65,7 +65,7 @@ export const AddStoryPage = ({
       .then((data) => {
         // Success! Move on to the full edit form
         history.push(
-          `/${feed?.name}/prospects/article/edit-and-approve/${data.data?.prospect.id}/`,
+          `/${feed?.name}/prospects/${data.data?.prospect.id}/edit-and-approve/`,
```
Comment (jpetto) [inline src/pages/AddStoryPage/AddStoryPage.tsx:68]: a little concerned about the `data.data?.` part here - if `data.data` is empty, we probably shouldn't push to the history, right?

**Review by jpetto — APPROVED** [review]
had one comment about a potential `null` value in a `history.push`, but we can address that at a later time to stop blocking here.

File: src/pages/AddStoryPage/AddStoryPage.tsx (hunk 3 above)
Comment (nina-py) [reply src/pages/AddStoryPage/AddStoryPage.tsx:68]: Good point, I've looked up Apollo types and it can potentially be null or undefined.

**Pull request merged**